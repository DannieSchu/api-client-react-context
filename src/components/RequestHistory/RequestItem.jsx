import React from 'react';
import PropTypes from 'prop-types';
import styles from './RequestItem.css';

const RequestItem = ({ method, url, onClick }) => (
  <section className={styles.RequestItem} onClick={onClick}>
    <h3>{method}</h3>
    <p>{url}</p>
  </section>
);

RequestItem.propTypes = {
  method: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired
};

export default RequestItem;
