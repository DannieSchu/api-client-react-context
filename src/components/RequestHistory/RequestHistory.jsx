import React from 'react';
import PropTypes from 'prop-types';
import RequestItem from './RequestItem.jsx';
import { weakKey } from '../../utilities/weakKey.jsx';
import styles from './RequestHistory.css';
import buttonStyle from '../../css/button.css';

const RequestHistory = ({ requests, onClick, onDelete }) => {
  const requestElements = requests.map(request => (
    <li key={weakKey(request)}>
      <RequestItem {...request} onClick={() => onClick(request)} />
    </li>
  ));
  
  return (
    <ul className={styles.RequestHistory}>
      {requestElements}
      <button className={buttonStyle.tealButton} onClick={onDelete}>Clear History</button>
    </ul>
  );
};

RequestHistory.propTypes = {
  onClick: PropTypes.func.isRequired,
  onDelete: PropTypes.func.isRequired,
  requests: PropTypes.arrayOf(PropTypes.shape({
    method: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    body: PropTypes.string,
  }))
};

export default RequestHistory;
