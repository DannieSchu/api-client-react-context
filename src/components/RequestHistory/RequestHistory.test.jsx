import React from 'react';
import { shallow } from 'enzyme';
import RequestHistory from './RequestHistory.jsx';

describe('RequestHistory', () => {
  it('matches a snapshot', () => {
    const requests = [{
      method: 'post', 
      url: 'https://jsonplaceholder.typicode.com/posts'
    }, {
      method: 'get', 
      url: 'https://jsonplaceholder.typicode.com/posts'
    }, {
      method: 'put', 
      url: 'https://jsonplaceholder.typicode.com/posts'
    }, {
      method: 'put', 
      url: 'https://jsonplaceholder.typicode.com/posts'
    }];

    const wrapper = shallow(
      <RequestHistory requests={requests} onClick={() => {}}onDelete={() => {}} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
