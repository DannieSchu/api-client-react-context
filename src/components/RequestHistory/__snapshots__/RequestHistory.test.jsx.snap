// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequestHistory matches a snapshot 1`] = `
<ul
  className="RequestHistory"
>
  <li
    key="weak-key-0"
  >
    <RequestItem
      method="post"
      onClick={[Function]}
      url="https://jsonplaceholder.typicode.com/posts"
    />
  </li>
  <li
    key="weak-key-1"
  >
    <RequestItem
      method="get"
      onClick={[Function]}
      url="https://jsonplaceholder.typicode.com/posts"
    />
  </li>
  <li
    key="weak-key-2"
  >
    <RequestItem
      method="put"
      onClick={[Function]}
      url="https://jsonplaceholder.typicode.com/posts"
    />
  </li>
  <li
    key="weak-key-3"
  >
    <RequestItem
      method="put"
      onClick={[Function]}
      url="https://jsonplaceholder.typicode.com/posts"
    />
  </li>
  <button
    className="tealButton"
    onClick={[Function]}
  >
    Clear History
  </button>
</ul>
`;
